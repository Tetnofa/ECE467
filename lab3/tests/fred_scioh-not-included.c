// Declare variables to test I/O functions without including scio.h
int x;
int y;
bool b1;
bool b2;

void f1() {
  // Test read functions (should produce errors due to missing `scio.h`)
  x = readInt();   // Error: readInt not declared
  y = readInt();   // Error: readInt not declared
  b1 = readBool(); // Error: readBool not declared
  b2 = readBool(); // Error: readBool not declared

  // Test write functions (should produce errors due to missing `scio.h`)
  writeInt(x);   // Error: writeInt not declared
  writeInt(y);   // Error: writeInt not declared
  writeBool(b1); // Error: writeBool not declared
  writeBool(b2); // Error: writeBool not declared

  // Test newLine function (should produce error due to missing `scio.h`)
  newLine(); // Error: newLine not declared

  // Nested I/O operations (should produce errors)
  writeInt(readInt());   // Error: both writeInt and readInt not declared
  writeBool(readBool()); // Error: both writeBool and readBool not declared
}
