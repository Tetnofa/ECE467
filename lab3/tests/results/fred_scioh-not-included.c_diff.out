0a1,75
> Program [useIO=0](2,0)
> 	Scalar Declaration	Int(2,0)
> 		Identifier[name:x](2,4)
> 	Scalar Declaration	Int(3,0)
> 		Identifier[name:y](3,4)
> 	Scalar Declaration	Bool(4,0)
> 		Identifier[name:b1](4,5)
> 	Scalar Declaration	Bool(5,0)
> 		Identifier[name:b2](5,5)
> 	Function[isProto=0]	Void(7,0)
> 		Identifier[name:f1](7,5)
> 		Scope(7,10)
> 			Assignment(9,2)
> 				Reference(9,2)
> 					Identifier[name:x](9,2)
> 				CallExpr(9,6)
> 					Identifier[name:readInt](9,6)
> 			Assignment(10,2)
> 				Reference(10,2)
> 					Identifier[name:y](10,2)
> 				CallExpr(10,6)
> 					Identifier[name:readInt](10,6)
> 			Assignment(11,2)
> 				Reference(11,2)
> 					Identifier[name:b1](11,2)
> 				CallExpr(11,7)
> 					Identifier[name:readBool](11,7)
> 			Assignment(12,2)
> 				Reference(12,2)
> 					Identifier[name:b2](12,2)
> 				CallExpr(12,7)
> 					Identifier[name:readBool](12,7)
> 			Expression Statement(15,2)
> 				CallExpr(15,2)
> 					Identifier[name:writeInt](15,2)
> 					Argument(15,11)
> 						Int Expression(15,11)
> 							Reference(15,11)
> 								Identifier[name:x](15,11)
> 			Expression Statement(16,2)
> 				CallExpr(16,2)
> 					Identifier[name:writeInt](16,2)
> 					Argument(16,11)
> 						Int Expression(16,11)
> 							Reference(16,11)
> 								Identifier[name:y](16,11)
> 			Expression Statement(17,2)
> 				CallExpr(17,2)
> 					Identifier[name:writeBool](17,2)
> 					Argument(17,12)
> 						Int Expression(17,12)
> 							Reference(17,12)
> 								Identifier[name:b1](17,12)
> 			Expression Statement(18,2)
> 				CallExpr(18,2)
> 					Identifier[name:writeBool](18,2)
> 					Argument(18,12)
> 						Int Expression(18,12)
> 							Reference(18,12)
> 								Identifier[name:b2](18,12)
> 			Expression Statement(21,2)
> 				CallExpr(21,2)
> 					Identifier[name:newLine](21,2)
> 			Expression Statement(24,2)
> 				CallExpr(24,2)
> 					Identifier[name:writeInt](24,2)
> 					Argument(24,11)
> 						CallExpr(24,11)
> 							Identifier[name:readInt](24,11)
> 			Expression Statement(25,2)
> 				CallExpr(25,2)
> 					Identifier[name:writeBool](25,2)
> 					Argument(25,12)
> 						CallExpr(25,12)
> 							Identifier[name:readBool](25,12)
